drivers/serial/uart_nrfx_uarte.c

```diff
@@ -674,6 +674,13 @@ static void endrx_isr(struct device *dev)
 
 	size_t rx_len = nrf_uarte_rx_amount_get(uarte)
 			- data->async->rx_offset;
+
+	data->async->rx_total_user_byte_cnt += rx_len;
+
+	if (!hw_rx_counting_enabled(data)) {
+		data->async->rx_cnt.cnt = data->async->rx_total_user_byte_cnt;
```

Comment:
It would be good to add a comment here that this is done to prevent using an incorrect (too low) value of `rx_cnt.cnt` (which may occur due to latencies in handling of the `RXDRDY` interrupt) in the computation of the amount of new bytes to be passed to the user callback in `rx_timeout` (this is not that obvious). And I think it is also worth mentioning that because the whole buffer has been filled we can be sure here that the above value (`rx_total_user_byte_cnt`) is the current total number of received bytes.